### ISSUE TITLE
Groups feedback: support all actors in cohorts

### ISSUE THREAD

**Issue by joesaunderson** [PostHog/posthog#8010]
## Is your feature request related to a problem?

When we first implemented Posthog, we used the user properties to indicate some group properties i.e. a user before looked like:

```
name: Joe Saunderson
role: ROLE_ADMIN
merchantId: 1234
merchantICPLevel: 1
```

```
name: John Doe
role: ROLE_USER
merchantId: 4321
merchantICPLevel: 3
```

We built cohorts using these properties, for example, admin users of ICP 1 merchants. 

We have now implemented groups in our Posthog instance, and the groups (merchant) related properties are no longer set on a user.

## Describe the solution you'd like

We'd like to be able to build cohorts based on both user properties and group properties, so in the above use case, I would expect the cohort builder UI to show:

`user.role = ROLE_ADMIN` AND `merchant.ICPLevel = 1`

## Describe alternatives you've considered

Dropping use of cohorts here, and adding filters to insights - which is very repetitive for insights that need these a lot.

## Additional context

CC @macobo for the groups feedback.

**Comment by macobo** [comment]
Mix-and-matching properties is not going to work currently due to groups not translating 1:N to people in all cases.

I wonder if allowing creating group cohorts would help? :thinking:

**Comment by joesaunderson** [comment]
> I wonder if allowing creating group cohorts would help? :thinking: 

I think it would, in this case, we could apply two cohorts to the filtering (admin users and ICP groups) to show the correct data in an insight.

**Comment by lsmith77** [comment]
ideally cohorts would have a “member of” criteria

**Comment by dmarticus** [comment]
Closing as a duplicate of #33096, which is the canonical issue for group-based cohorts. The original ask here — building cohorts that mix person and group properties — is part of that broader scope (group cohorts unblock the "two cohorts applied together" pattern @joesaunderson described). Please follow #33096 for updates.
